firefox-ios/Client/Frontend/Home/Homepage Rebuild/TopSites/TopSitesDimensionImplementation.swift

```diff
@@ -0,0 +1,22 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+struct TopSitesDimensionImplementation {
+    /// Get the number of tiles (top sites) per row the user will see. This depends on the UI interface the user has.
+    /// - Parameter availableWidth: available width size depending on device
+    /// - Parameter leadingInset: padding for top site section
+    /// - Parameter cellWidth: width of individual top site tiles
+    /// - Returns: The number of tiles per row the user will see, which is based on layout.
+    func getNumberOfTilesPerRow(availableWidth: CGFloat, leadingInset: CGFloat, cellWidth: CGFloat) -> Int {
+        var availableWidth = availableWidth - leadingInset * 2
```

Comment:
copied this over from legacy, but changed the params for simplicity + clarity

Old implementation:
```
/// Get the number of tiles per row the user will see. This depends on the UI interface the user has.
    /// - Parameter interface: Tile number is based on layout, this param contains the parameters
    ///                        needed to computer the tile number
    /// - Returns: The number of tiles per row the user will see
    private func getNumberOfTilesPerRow(for interface: TopSitesUIInterface) -> Int {
        let cellWidth = TopSitesViewModel.UX.cellEstimatedSize.width
        let leadingInset = HomepageViewModel.UX.leadingInset(traitCollection: interface.trait,
                                                             interfaceIdiom: interface.interfaceIdiom)
        var availableWidth = interface.availableWidth - leadingInset * 2
        var numberOfTiles = 0

        while availableWidth > cellWidth {
            numberOfTiles += 1
            availableWidth = availableWidth - cellWidth - TopSitesViewModel.UX.cardSpacing
        }
        return numberOfTiles < TopSitesViewModel.UX.minCards ? TopSitesViewModel.UX.minCards : numberOfTiles
    }
    ```